src/Cache.cpp

```diff
@@ -580,7 +583,10 @@ Cache::storeSecret(const std::string &name, const std::string &secret)
         QKeychain::WritePasswordJob job(QCoreApplication::applicationName());
         job.setAutoDelete(false);
         job.setInsecureFallback(true);
-        job.setKey(QString::fromStdString(name));
+        job.setKey(
+          "matrix." +
+          QString(QCryptographicHash::hash(localUserId_.toUtf8(), QCryptographicHash::Sha256)) +
```

Comment:
Please use the profile here, not the userid. Otherwise logout deletes the wrong secrets.